src/test/scala/com/cognite/sdk/scala/v1/TransformationsTest.scala

```diff
@@ -0,0 +1,208 @@
+    resCreates.size shouldBe transformationsToCreate.size
+    resCreates.map(_.name) should contain theSameElementsAs transformationsToCreate.map(_.name)
+    resCreates.map(_.query) should contain theSameElementsAs transformationsToCreate.flatMap(
+      _.query
+    )
+    resCreates.map(_.destination) should contain theSameElementsAs transformationsToCreate.flatMap(
+      _.destination
+    )
+    resCreates.map(_.externalId) should contain theSameElementsAs transformationsToCreate.map(
+      _.externalId
+    )
+
+    val externalIds = transformationsToCreate.map(_.externalId)
+    retryWithExpectedResult[Seq[TransformationRead]](
+      {
+        client.transformations.delete(externalIds.map(CogniteExternalId(_)), true).unsafeRunSync()
+        client.transformations.retrieveByExternalIds(externalIds, true).unsafeRunSync()
+      },
+      t => t shouldBe empty
+    )
+  }
+
+  it should "retrieve transformations by ids or externalIds" in {
+    val transformationsToCreate = (0 to 3).map { i =>
+      val uniquePrefix = shortRandomUUID()
+      TransformationCreate(
+        s"$uniquePrefix-transformation-sdk-test",
+        Some(s"select ${i.toString}"),
+        Some(GeneralDataSource("events")),
+        conflictMode = Some("upsert"),
+        Some(false),
+        sourceOidcCredentials = Some(credentials),
+        destinationOidcCredentials = Some(credentials),
+        externalId = s"$uniquePrefix-transformation-sdk-test",
+        ignoreNullFields = true,
+        dataSetId = None
+      )
+    }
+    val resCreated =
+      client.transformations.createItems(Items(transformationsToCreate)).unsafeRunSync()
+    val externalIds = transformationsToCreate.map(_.externalId)
+    val internalIds = resCreated.map(_.id)
+
+    try {
+      val resByIds = client.transformations.retrieveByIds(internalIds, true).unsafeRunSync()
+      resByIds.size shouldBe transformationsToCreate.size
+
+      val resByExternalIds =
+        client.transformations.retrieveByExternalIds(externalIds, true).unsafeRunSync()
+      resByExternalIds.size shouldBe transformationsToCreate.size
+    } finally
+      try
+        client.transformations.delete(externalIds.map(CogniteExternalId(_)), true).unsafeRunSync()
+      catch {
+        case NonFatal(_) => // ignore
+      }
+  }
+
+  it should "filter transformations" in {
+    val uniquePrefix = shortRandomUUID()
+    val transformationsToCreate = (0 to 3).map { i =>
+      TransformationCreate(
+        s"$uniquePrefix-transformation-sdk-test-${i.toString}",
+        Some(s"select abc ${i.toString}"),
+        Some(GeneralDataSource("events")),
+        conflictMode = Some("upsert"),
+        Some(false),
+        sourceOidcCredentials = Some(credentials),
+        destinationOidcCredentials = Some(credentials),
+        externalId = s"$uniquePrefix-transformation-sdk-test-${i.toString}",
+        ignoreNullFields = true,
+        dataSetId = Some(216250735038513L)
```

Comment:
Please move this to a test-level `val` and give it a descriptive name.

Reply:
done :)